rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp

```diff
@@ -138,40 +138,39 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
             y1 =  (original_image_height - crop_height) / 2;
         }
     }
-    
-   // std::cout<<"Fused Crop Decoder <x,y, w, h>: " << x1 << " " << y1 << " " << crop_width << " " << crop_height << std::endl;
+    // Handles the case where generated crop width / crop height can be greater than the max decoded width & max decoded height set by the user.
+    if(crop_width > max_decoded_width)
+        crop_width = max_decoded_width;
+    if(crop_height > max_decoded_height)
+        crop_height = max_decoded_height;
     //TODO : Turbo Jpeg supports multiple color packing and color formats, add more as an option to the API TJPF_RGB, TJPF_BGR, TJPF_RGBX, TJPF_BGRX, TJPF_RGBA, TJPF_GRAY, TJPF_CMYK , ...
-    if( tjDecompress2_partial(m_jpegDecompressor,
-                      input_buffer,
-                      input_size,
-                      output_buffer,
-                      max_decoded_width,
-                      max_decoded_width * planes,
-                      max_decoded_height,
-                      tjpf,
-                      TJFLAG_FASTDCT, &x1_diff, &crop_width_diff,
-		                  x1, y1, crop_width, crop_height) != 0)
+    if(tjDecompress2_partial(m_jpegDecompressor,
+                      	     input_buffer,
+                             input_size,
+                             output_buffer,
+                             max_decoded_width,
+                             max_decoded_width * planes,
+                             max_decoded_height,
+                             tjpf,
+                             TJFLAG_FASTDCT, &x1_diff, &crop_width_diff,
+		             x1, y1, crop_width, crop_height) != 0)
 
     {
         WRN("Jpeg image decode failed " + STR(tjGetErrorStr2(m_jpegDecompressor)))
         return Status::CONTENT_DECODE_FAILED;
     }
 
     if (x1 != x1_diff) {
-        //std::cout << "x_off changed by tjpeg decoder " << x1 << " " << x1_diff << std::endl;
         unsigned char *src_ptr_temp, *dst_ptr_temp;
-        unsigned int elements_in_row = max_decoded_width * planes;
         unsigned int elements_in_crop_row = crop_width * planes;
-        //unsigned int remainingElements =  elements_in_row - elements_in_crop_row;
-        unsigned int xoffs = (x1-x1_diff) * planes;   // in case x1 gets adjusted by tjpeg decoder
-        src_ptr_temp = output_buffer + xoffs;
+        unsigned int xoffs = (x1 - x1_diff) * planes;   // in case x1 gets adjusted by tjpeg decoder
+        src_ptr_temp = output_buffer;
         dst_ptr_temp = output_buffer;
         for (unsigned int i = 0; i < crop_height; i++)
         {
             memcpy(dst_ptr_temp, src_ptr_temp + xoffs, elements_in_crop_row * sizeof(unsigned char));
-            //memset(dst_ptr_temp + elements_in_crop_row, 0, remainingElements * sizeof(unsigned char));
-            src_ptr_temp +=  elements_in_row;
-            dst_ptr_temp +=  elements_in_row;
+            src_ptr_temp +=  xoffs + elements_in_crop_row;
```

Comment:
Not sure if this is the right math. The stride is elements_in_row which may not be same as (elements_in_row)